Rename isHTMLSpace and move it to WTF

annevk opened the pull request:
#### 7f7aade4cf4a5c2688604f64e7b0f901771a3241
<pre>
Rename isHTMLSpace and move it to WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=255840">https://bugs.webkit.org/show_bug.cgi?id=255840</a>
rdar://108419120

Reviewed by Darin Adler.

Rename isHTMLSpace to isASCIIWhitespace to align it with the Infra standard, make it a constexpr
following precedent set in c650d41f8945032a348d7ffd4e11c9ca2dd75ea7, and move it to WTF to make it
more easily usable throughout WebKit. Put it in ASCIICType.h because based on
169adaafef454897deb4507150d2708430e4de01 it seemed like a reasonable place for ASCII-related
functionality.

Likewise for isNotHTMLSpace though non-predicate instances are replaced with !isASCIIWhitespace.

At the same time rename isASCIISpace to isUnicodeCompatibleASCIIWhitespace to make it clearer this
does not match the typical web standard usage and has to be somewhat carefully considered. Auditing
of existing usage will continue in <a href="https://bugs.webkit.org/show_bug.cgi?id=255467">https://bugs.webkit.org/show_bug.cgi?id=255467</a> and dependencies.

* Source/JavaScriptCore/bytecode/ReduceWhitespace.cpp:
(JSC::reduceWhitespace):
* Source/JavaScriptCore/runtime/ConfigFile.cpp:
(JSC::ConfigFileScanner::fillBufferIfNeeded):
(JSC::ConfigFile::parse):
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::isSeparator):
* Source/JavaScriptCore/tools/FunctionOverrides.cpp:
(JSC::hasDisallowedCharacters):
* Source/WTF/wtf/ASCIICType.h:
(WTF::isASCIIWhitespace):
(WTF::isUnicodeCompatibleASCIIWhitespace):
(WTF::isNotASCIIWhitespace):
(WTF::isASCIISpace): Deleted.
* Source/WTF/wtf/DateMath.cpp:
(WTF::skipSpacesAndComments):
(WTF::parseDateFromNullTerminatedCharacters):
* Source/WTF/wtf/text/Base64.cpp:
(WTF::base64DecodeInternal):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::isSpaceOrNewline):
(WTF::isUnicodeWhitespace):
* Source/WTF/wtf/text/StringToIntegerConversion.h:
(WTF::parseInteger):
* Source/WTF/wtf/text/StringView.cpp:
(WTF::StringView::stripWhiteSpace const):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::toDoubleType):
* Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp:
(WebCore::MediaRecorderProvider::isSupported):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::traverseToOffsetInRange):
* Source/WebCore/accessibility/AccessibilityList.cpp:
(WebCore::AccessibilityList::childHasPseudoVisibleListItemMarkers):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::objectInclusionFromAltText):
(WebCore::AccessibilityRenderObject::computeAccessibilityIsIgnored const):
* Source/WebCore/contentextensions/Term.h:
(WebCore::ContentExtensions::Term::toString const):
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::attributeValueMatches):
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::parseColorIntOrPercentage):
(WebCore::parseAlphaValue):
* Source/WebCore/css/parser/CSSTokenizer.cpp:
(WebCore::CSSTokenizer::consumeURLToken):
(WebCore::CSSTokenizer::consumeSingleWhitespaceIfNext):
* Source/WebCore/css/parser/CSSTokenizerInputStream.cpp:
(WebCore::CSSTokenizerInputStream::advanceUntilNonWhitespace):
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::constructFragmentsInternal):
(WebCore::SelectorCompiler::attributeValueSpaceSeparatedListContains):
* Source/WebCore/dom/Document.cpp:
(WebCore::canonicalizedTitle):
(WebCore::processColorSchemeString):
* Source/WebCore/dom/Position.cpp:
(WebCore::Position::leadingWhitespacePosition const):
(WebCore::Position::trailingWhitespacePosition const):
* Source/WebCore/dom/SecurityContext.cpp:
(WebCore::SecurityContext::parseSandboxPolicy):
* Source/WebCore/dom/ShadowRoot.cpp:
(WebCore::parsePartMapping):
* Source/WebCore/dom/SpaceSplitString.cpp:
(WebCore::tokenizeSpaceSplitString):
* Source/WebCore/editing/TextManipulationController.cpp:
(WebCore::isNotSpace):
(WebCore::TextManipulationController::parse):
* Source/WebCore/html/DOMTokenList.cpp:
(WebCore::tokenContainsHTMLSpace):
(WebCore::DOMTokenList::updateTokensFromAttributeValue):
* Source/WebCore/html/FeaturePolicy.cpp:
(WebCore::processOriginItem):
(WebCore::updateList):
(WebCore::FeaturePolicy::parse):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::parseLegacyColorValue):
* Source/WebCore/html/HTMLFontElement.cpp:
(WebCore::parseFontSize):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::extractMIMETypeFromTypeAttributeForLookup):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::parseAcceptAttribute):
* Source/WebCore/html/HTMLObjectElement.cpp:
(WebCore::HTMLObjectElement::hasFallbackContent const):
(WebCore::preventsParentObjectFromExposure):
* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::text const):
(WebCore::HTMLOptionElement::value const):
(WebCore::HTMLOptionElement::label const):
(WebCore::HTMLOptionElement::displayLabel const):
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
(WebCore::CSSPreloadScanner::tokenize):
(WebCore::parseCSSStringOrURL):
(WebCore::hasValidImportConditions):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::isCharAfterTagNameOrAttribute):
(WebCore::isCharAfterUnquotedAttribute):
(WebCore::HTMLFastPathParser::scanTagName):
(WebCore::HTMLFastPathParser::scanAttributeValue):
(WebCore::HTMLFastPathParser::scanEscapedAttributeValue):
(WebCore::HTMLFastPathParser::parseAttributes):
(WebCore::HTMLFastPathParser::parseContainerElement):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.cpp:
(WebCore::HTMLMetaCharsetParser::encodingFromMetaAttributes):
* Source/WebCore/html/parser/HTMLParserIdioms.cpp:
(WebCore::stripLeadingAndTrailingHTMLSpaces):
(WebCore::parseHTMLIntegerInternal):
(WebCore::isHTMLSpaceOrDelimiter):
(WebCore::parseHTTPRefreshInternal):
(WebCore::parseHTMLDimensionNumber):
* Source/WebCore/html/parser/HTMLParserIdioms.h:
(WebCore::isHTMLSpaceOrComma):
(WebCore::isHTMLSpaceButNotLineBreak):
(WebCore::isHTMLSpace): Deleted.
(WebCore::isNotHTMLSpace): Deleted.
* Source/WebCore/html/parser/HTMLPreloadScanner.cpp:
(WebCore::TokenPreloadScanner::StartTagScanner::processImageAndScriptAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::processAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::setURLToLoadAllowingReplacement):
* Source/WebCore/html/parser/HTMLSrcsetParser.cpp:
(WebCore::tokenizeDescriptors):
(WebCore::parseImageCandidatesFromSrcsetAttribute):
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::isAllWhitespace):
(WebCore::isAllWhitespaceOrReplacementCharacters):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::skipLeadingWhitespace):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::takeLeadingWhitespace):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::skipLeadingNonWhitespace):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::takeRemainingWhitespace):
* Source/WebCore/html/parser/ParsingUtilities.h:
(WebCore::isNotASCIISpace):
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::setRegionSettings):
(WebCore::VTTRegion::parseSettingValue):
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::hasRequiredFileIdentifier):
(WebCore::WebVTTParser::collectTimingsAndSettings):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::StyleSheetHandler::setRuleHeaderEnd):
(WebCore::fixUnparsedProperties):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::containsOnlyHTMLWhitespace):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::extractContentLanguageFromHeader):
* Source/WebCore/loader/SubresourceIntegrity.cpp:
(WebCore::splitOnSpaces):
* Source/WebCore/mathml/MathMLMencloseElement.cpp:
(WebCore::MathMLMencloseElement::parseNotationAttribute):
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::parseMathMLLength):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::incrementVisuallyNonEmptyCharacterCount):
* Source/WebCore/page/WindowFeatures.cpp:
(WebCore::isSeparator):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::didReceiveHeader):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::isDirectiveValueCharacter):
(WebCore::ContentSecurityPolicyDirectiveList::parseDirective):
(WebCore::ContentSecurityPolicyDirectiveList::parseReportURI):
(WebCore::ContentSecurityPolicyDirectiveList::parseReportTo):
* Source/WebCore/page/csp/ContentSecurityPolicyMediaListDirective.cpp:
(WebCore::isMediaTypeCharacter):
(WebCore::ContentSecurityPolicyMediaListDirective::parse):
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.cpp:
(WebCore::isSourceCharacter):
(WebCore::isSourceListNone):
(WebCore::ContentSecurityPolicySourceList::parse):
* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::threadCPUUsage):
* Source/WebCore/platform/ContentType.cpp:
(WebCore::ContentType::parameter const):
(WebCore::splitParameters):
* Source/WebCore/platform/graphics/ComplexTextController.cpp:
(WebCore::shouldSynthesize):
* Source/WebCore/rendering/LegacyLineLayout.cpp:
(WebCore::endsWithHTMLSpaces):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::isVisibleRenderText):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::calculateClipRects const):
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::trimmedPreferredWidths):
* Source/WebCore/rendering/TextPainter.cpp:
(WebCore::TextPainter::paintTextOrEmphasisMarks):
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::updateRenderTree):
(WebCore::RenderTreeUpdater::textRendererIsNeeded):
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolveComposedTree):
* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::SVGAnimationElement::attributeContainsJavaScriptURL const):
(WebCore::SVGAnimationElement::attributeChanged):
* Source/WebGPU/WGSL/Lexer.cpp:
(WGSL::Lexer&lt;T&gt;::skipWhitespaceAndComments):
* Source/WebKit/Shared/linux/WebMemorySamplerLinux.cpp:
(WebKit::nextToken):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::getPlatformEditorState const):

Canonical link: <a href="https://commits.webkit.org/263290@main">https://commits.webkit.org/263290@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bc89fd18718e034d232d73111292da0486d06942

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4158 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4272 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4390 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5622 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4403 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4147 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4374 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4237 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/4626 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4217 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4392 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3748 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5615 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1888 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3725 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/5906 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/3452 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3715 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3790 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5309 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/3947 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4193 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3397 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4251 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3706 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3723 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1041 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1019 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/7828 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/4347 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3988 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1159 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/dom/SpaceSplitString.cpp:
```diff
@@ -34,12 +33,12 @@ template <typename CharacterType, typename TokenProcessor>
             break;
         unsigned end = start + 1;
-        while (end < length && isNotHTMLSpace(characters[end]))
+        while (end < length && isNotASCIIWhitespace(characters[end]))
```
I considered replacing instances such as this one with `!isASCIIWhitespace` leaving `isNotASCIIWhitespace` solely for when you need a callback. We're not exactly consistent about this. Advice appreciated.

darinadler replied:
I agree with this and think you should do what you considered. The reason `isNotASCIIWhitespace` exists is so we can use it as a predicate, which comes up mainly because of our `skip` function templates. Outside of that, I don’t think we should use it. The counter argument is that code using the “not”-named function is easier to read, but I think C  programmers learn to read code with `!` in it.

darinadler commented on Source/WTF/wtf/ASCIICType.h:
```diff
@@ -49,7 +49,9 @@ template<typename CharacterType> constexpr bool isASCIIHexDigit(CharacterType);
-template<typename CharacterType> constexpr bool isASCIISpace(CharacterType);
+template<typename CharacterType> constexpr bool isASCIIWhitespace(CharacterType);
+template<typename CharacterType> constexpr bool isNotASCIIWhitespace(CharacterType);
+template<typename CharacterType> constexpr bool isCCompatibleASCIIWhitespace(CharacterType);
```
I don’t think "C compatible" is the ideal phrase here. The broader version of whitespace that includes U+000B is:

1) Consistent with Unicode’s whitespace definition.
2) Consistent with the "C" locale behavior of the C standard library `isspace` function.

The second doesn’t really say "C compatible" to me. I would suggest a name more like `isTraditionalASCIIWhitespace` or `isOldFashionedASCIIWhitespace` or `isUnicodeCompatibleASCIIWhitespace`. I don’t love any of those, but I think they are all clearer than "C compatible".

annevk replied:
I went with `isUnicodeCompatibleASCIIWhitespace` for now. Although for 8-bit strings we probably need a Latin1-compatible definition for it to be useful.

darinadler replied:
We have that elsewhere for Latin-1. I'll find it for you.

darinadler commented on Source/WTF/wtf/ASCIICType.h:
```diff
@@ -49,7 +49,9 @@ template<typename CharacterType> constexpr bool isASCIIHexDigit(CharacterType);
 template<typename CharacterType> constexpr bool isASCIIPrintable(CharacterType);
-template<typename CharacterType> constexpr bool isASCIISpace(CharacterType);
+template<typename CharacterType> constexpr bool isASCIIWhitespace(CharacterType);
+template<typename CharacterType> constexpr bool isNotASCIIWhitespace(CharacterType);
```
I suggest putting this in a separate paragraph and documenting why we have an `isNotASCIIWhitespace` function: It’s common enough to want this as a predicate that we include it.

darinadler reviewed (approved):
Very nice. Great to have the terminology in `ASCIICType.h` exactly match the infrastructure specification.

I think we could slightly improve with a different name for the legacy function.

annevk committed:
Rename isHTMLSpace and move it to WTF
https://bugs.webkit.org/show_bug.cgi?id=255840
rdar://108419120

Reviewed by Darin Adler.

Rename isHTMLSpace to isASCIIWhitespace to align it with the Infra standard, make it a constexpr
following precedent set in c650d41f8945032a348d7ffd4e11c9ca2dd75ea7, and move it to WTF to make it
more easily usable throughout WebKit. Put it in ASCIICType.h because based on
169adaafef454897deb4507150d2708430e4de01 it seemed like a reasonable place for ASCII-related
functionality.

Likewise for isNotHTMLSpace though non-predicate instances are replaced with !isASCIIWhitespace.

At the same time rename isASCIISpace to isUnicodeCompatibleASCIIWhitespace to make it clearer this
does not match the typical web standard usage and has to be somewhat carefully considered. Auditing
of existing usage will continue in https://bugs.webkit.org/show_bug.cgi?id=255467 and dependencies.

* Source/JavaScriptCore/bytecode/ReduceWhitespace.cpp:
(JSC::reduceWhitespace):
* Source/JavaScriptCore/runtime/ConfigFile.cpp:
(JSC::ConfigFileScanner::fillBufferIfNeeded):
(JSC::ConfigFile::parse):
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::isSeparator):
* Source/JavaScriptCore/tools/FunctionOverrides.cpp:
(JSC::hasDisallowedCharacters):
* Source/WTF/wtf/ASCIICType.h:
(WTF::isASCIIWhitespace):
(WTF::isUnicodeCompatibleASCIIWhitespace):
(WTF::isNotASCIIWhitespace):
(WTF::isASCIISpace): Deleted.
* Source/WTF/wtf/DateMath.cpp:
(WTF::skipSpacesAndComments):
(WTF::parseDateFromNullTerminatedCharacters):
* Source/WTF/wtf/text/Base64.cpp:
(WTF::base64DecodeInternal):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::isSpaceOrNewline):
(WTF::isUnicodeWhitespace):
* Source/WTF/wtf/text/StringToIntegerConversion.h:
(WTF::parseInteger):
* Source/WTF/wtf/text/StringView.cpp:
(WTF::StringView::stripWhiteSpace const):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::toDoubleType):
* Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp:
(WebCore::MediaRecorderProvider::isSupported):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::traverseToOffsetInRange):
* Source/WebCore/accessibility/AccessibilityList.cpp:
(WebCore::AccessibilityList::childHasPseudoVisibleListItemMarkers):
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::objectInclusionFromAltText):
(WebCore::AccessibilityRenderObject::computeAccessibilityIsIgnored const):
* Source/WebCore/contentextensions/Term.h:
(WebCore::ContentExtensions::Term::toString const):
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::attributeValueMatches):
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::parseColorIntOrPercentage):
(WebCore::parseAlphaValue):
* Source/WebCore/css/parser/CSSTokenizer.cpp:
(WebCore::CSSTokenizer::consumeURLToken):
(WebCore::CSSTokenizer::consumeSingleWhitespaceIfNext):
* Source/WebCore/css/parser/CSSTokenizerInputStream.cpp:
(WebCore::CSSTokenizerInputStream::advanceUntilNonWhitespace):
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::constructFragmentsInternal):
(WebCore::SelectorCompiler::attributeValueSpaceSeparatedListContains):
* Source/WebCore/dom/Document.cpp:
(WebCore::canonicalizedTitle):
(WebCore::processColorSchemeString):
* Source/WebCore/dom/Position.cpp:
(WebCore::Position::leadingWhitespacePosition const):
(WebCore::Position::trailingWhitespacePosition const):
* Source/WebCore/dom/SecurityContext.cpp:
(WebCore::SecurityContext::parseSandboxPolicy):
* Source/WebCore/dom/ShadowRoot.cpp:
(WebCore::parsePartMapping):
* Source/WebCore/dom/SpaceSplitString.cpp:
(WebCore::tokenizeSpaceSplitString):
* Source/WebCore/editing/TextManipulationController.cpp:
(WebCore::isNotSpace):
(WebCore::TextManipulationController::parse):
* Source/WebCore/html/DOMTokenList.cpp:
(WebCore::tokenContainsHTMLSpace):
(WebCore::DOMTokenList::updateTokensFromAttributeValue):
* Source/WebCore/html/FeaturePolicy.cpp:
(WebCore::processOriginItem):
(WebCore::updateList):
(WebCore::FeaturePolicy::parse):
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::parseLegacyColorValue):
* Source/WebCore/html/HTMLFontElement.cpp:
(WebCore::parseFontSize):
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::extractMIMETypeFromTypeAttributeForLookup):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::parseAcceptAttribute):
* Source/WebCore/html/HTMLObjectElement.cpp:
(WebCore::HTMLObjectElement::hasFallbackContent const):
(WebCore::preventsParentObjectFromExposure):
* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::text const):
(WebCore::HTMLOptionElement::value const):
(WebCore::HTMLOptionElement::label const):
(WebCore::HTMLOptionElement::displayLabel const):
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
(WebCore::CSSPreloadScanner::tokenize):
(WebCore::parseCSSStringOrURL):
(WebCore::hasValidImportConditions):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::isCharAfterTagNameOrAttribute):
(WebCore::isCharAfterUnquotedAttribute):
(WebCore::HTMLFastPathParser::scanTagName):
(WebCore::HTMLFastPathParser::scanAttributeValue):
(WebCore::HTMLFastPathParser::scanEscapedAttributeValue):
(WebCore::HTMLFastPathParser::parseAttributes):
(WebCore::HTMLFastPathParser::parseContainerElement):
* Source/WebCore/html/parser/HTMLMetaCharsetParser.cpp:
(WebCore::HTMLMetaCharsetParser::encodingFromMetaAttributes):
* Source/WebCore/html/parser/HTMLParserIdioms.cpp:
(WebCore::stripLeadingAndTrailingHTMLSpaces):
(WebCore::parseHTMLIntegerInternal):
(WebCore::isHTMLSpaceOrDelimiter):
(WebCore::parseHTTPRefreshInternal):
(WebCore::parseHTMLDimensionNumber):
* Source/WebCore/html/parser/HTMLParserIdioms.h:
(WebCore::isHTMLSpaceOrComma):
(WebCore::isHTMLSpaceButNotLineBreak):
(WebCore::isHTMLSpace): Deleted.
(WebCore::isNotHTMLSpace): Deleted.
* Source/WebCore/html/parser/HTMLPreloadScanner.cpp:
(WebCore::TokenPreloadScanner::StartTagScanner::processImageAndScriptAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::processAttribute):
(WebCore::TokenPreloadScanner::StartTagScanner::setURLToLoadAllowingReplacement):
* Source/WebCore/html/parser/HTMLSrcsetParser.cpp:
(WebCore::tokenizeDescriptors):
(WebCore::parseImageCandidatesFromSrcsetAttribute):
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::isAllWhitespace):
(WebCore::isAllWhitespaceOrReplacementCharacters):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::skipLeadingWhitespace):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::takeLeadingWhitespace):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::skipLeadingNonWhitespace):
(WebCore::HTMLTreeBuilder::ExternalCharacterTokenBuffer::takeRemainingWhitespace):
* Source/WebCore/html/parser/ParsingUtilities.h:
(WebCore::isNotASCIISpace):
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::setRegionSettings):
(WebCore::VTTRegion::parseSettingValue):
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::hasRequiredFileIdentifier):
(WebCore::WebVTTParser::collectTimingsAndSettings):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::StyleSheetHandler::setRuleHeaderEnd):
(WebCore::fixUnparsedProperties):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::containsOnlyHTMLWhitespace):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::extractContentLanguageFromHeader):
* Source/WebCore/loader/SubresourceIntegrity.cpp:
(WebCore::splitOnSpaces):
* Source/WebCore/mathml/MathMLMencloseElement.cpp:
(WebCore::MathMLMencloseElement::parseNotationAttribute):
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::parseMathMLLength):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::incrementVisuallyNonEmptyCharacterCount):
* Source/WebCore/page/WindowFeatures.cpp:
(WebCore::isSeparator):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::didReceiveHeader):
* Source/WebCore/page/csp/ContentSecurityPolicyDirectiveList.cpp:
(WebCore::isDirectiveValueCharacter):
(WebCore::ContentSecurityPolicyDirectiveList::parseDirective):
(WebCore::ContentSecurityPolicyDirectiveList::parseReportURI):
(WebCore::ContentSecurityPolicyDirectiveList::parseReportTo):
* Source/WebCore/page/csp/ContentSecurityPolicyMediaListDirective.cpp:
(WebCore::isMediaTypeCharacter):
(WebCore::ContentSecurityPolicyMediaListDirective::parse):
* Source/WebCore/page/csp/ContentSecurityPolicySourceList.cpp:
(WebCore::isSourceCharacter):
(WebCore::isSourceListNone):
(WebCore::ContentSecurityPolicySourceList::parse):
* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::threadCPUUsage):
* Source/WebCore/platform/ContentType.cpp:
(WebCore::ContentType::parameter const):
(WebCore::splitParameters):
* Source/WebCore/platform/graphics/ComplexTextController.cpp:
(WebCore::shouldSynthesize):
* Source/WebCore/rendering/LegacyLineLayout.cpp:
(WebCore::endsWithHTMLSpaces):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::isVisibleRenderText):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::calculateClipRects const):
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::trimmedPreferredWidths):
* Source/WebCore/rendering/TextPainter.cpp:
(WebCore::TextPainter::paintTextOrEmphasisMarks):
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
(WebCore::RenderTreeUpdater::updateRenderTree):
(WebCore::RenderTreeUpdater::textRendererIsNeeded):
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolveComposedTree):
* Source/WebCore/svg/SVGAnimationElement.cpp:
(WebCore::SVGAnimationElement::attributeContainsJavaScriptURL const):
(WebCore::SVGAnimationElement::attributeChanged):
* Source/WebGPU/WGSL/Lexer.cpp:
(WGSL::Lexer<T>::skipWhitespaceAndComments):
* Source/WebKit/Shared/linux/WebMemorySamplerLinux.cpp:
(WebKit::nextToken):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::getPlatformEditorState const):

Canonical link: https://commits.webkit.org/263290@main
